### PR TITLE
doc: removed reference to pump and added description on http2SessionTimeout

### PR DESCRIPTION
**Pull request by mccare**
* removed reference to pump, since it seems like pump is not used internally in request handling (its a dev dependency and can only be found in tests)
* added some description why the current default http2SessionTimeout was chosen (DoS mitigation)

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Server.md [hunk 1]
```diff
@@ -454,8 +454,12 @@ const fastify = require('fastify')({
 Set a default
 [timeout](https://nodejs.org/api/http2.html#http2_http2session_settimeout_msecs_callback) to every incoming HTTP/2 session. The session will be closed on the timeout. Default: `5000` ms.
 
+
+
```
Comment (mcollina) [inline docs/Server.md:458]:
```suggestion

```

**Review by mcollina — COMMENTED** [review]
Thanks!

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: docs/Server.md [hunk 2]
```diff
@@ -454,8 +454,10 @@ const fastify = require('fastify')({
 Set a default
 [timeout](https://nodejs.org/api/http2.html#http2_http2session_settimeout_msecs_callback) to every incoming HTTP/2 session. The session will be closed on the timeout. Default: `5000` ms.
 
-Note that this is needed to offer the graceful "close" experience when
-using HTTP/2. Node core defaults this to `0`.
+Note that this is needed to offer the graceful "close" experience when using HTTP/2. 
+The low default has been chosen to mitigate denial of service attacks. 
+When the server is behind a load balancer or can scale automatically this value can be 
+increased to fit the use case. Node core defaults this to `0`. `
```
Comment (L2jLiga) [inline docs/Server.md:460]: I think here's small typo:

```suggestion
increased to fit the use case. Node core defaults this to `0`.
```

**Pull request merged**